### DISCUSSION TITLE
Auto-Start on User Login/Boot

### DISCUSSION THREAD

**Discussion by jdheeter** [tauri-apps/tauri#2576] [Q&A] [answered]
Is there any guide for Tauri apps to register with the OS to start with boot?

There is an npm module designed to work for Electron: https://www.npmjs.com/package/auto-launch

I think a proper implementation would make sense for Tauri in Rust. Can someone point me in the right direction?

**Comment by FabianLars** [comment, marked as answer]
Edit: Seems like we want to integrate it into tauri directly. Tracking issue: #2589

afaik there is no guide or example, not even a general crate.
But it should be quite straight forward to implement it in rust, as long as you don't need App Store compliance.
Basically you just have to copy a small file to a specific folder:

For example this one more or less tries to translate the functionality of the referenced package into rust, so it should give general ideas on how to do stuff: https://github.com/vn971/timeplot/blob/master/src/autostart.rs (doesn't look perfect, but it's the first thing i found :P )

Sounds like a nice opportunity for a community plugin to me.

**Reply by mainrs** [reply]
It is also possible to do it with App Store compliance. Each platform has their own specifics. But a custom wix template would work for windows. On Mac you'd have to register it inside your pdlist iirc and the packager can set the application to auto start. Linux had a ton of different options on how to do it, depending on what type of packaging you use.

**Reply by jdheeter** [reply]
Thanks for the pointers. I agree there is good potential here for a Tauri plugin. I'll report back on my progress as I've been working on implementing this functionality at a basic level.

**Comment by jdheeter** [comment]
For anyone who is looking for a solution. Here is our first revision, working on Windows, Macos, Linux. It's not tested in production yet and not properly wrapped up as a tauri plugin but if anyone in the community would like to work on an autolaunch Tauri plugin this code would be good to study for an integration.

https://github.com/subspace/desktop-farmer/blob/main/src/lib/autoLauncher.ts
https://github.com/subspace/desktop-farmer/blob/main/src-tauri/src/main.rs

**Comment by naps62** [comment]
Hey everyone

I just solved this rather quickly by using the [`auto-launch`](https://crates.io/crates/auto-launch) crate

My changes (tested on Linux & macos): https://github.com/iron-wallet/iron/pull/519

Keep in mind that for linux, this has various levels of support. it works by creating a file in `~/.config/autostart/`. My own desktop manager is hyprland, which explicitly does not care about this directory. Other tiling managers often behave similarly. In those cases, users need to set up autostart themselves (and they usually know how to)

**Comment by Kerinlin** [comment]
https://github.com/tauri-apps/plugins-workspace/tree/v1/plugins/autostart

**Reply by aqibhassanzeb** [reply]
I attempted to configure the code for Tauri using the URL: https://github.com/tauri-apps/plugins-workspace/tree/v1/plugins/autostart, but unfortunately, it's not functioning as expected in my Tauri + React.js setup. @Kerinlin 

@mainrs
